src/main/resources/hudson/tasks/junit/pipeline/JUnitPipelineStep.groovy

```diff
@@ -0,0 +1,47 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.tasks.junit.pipeline
+
+import org.jenkinsci.plugins.workflow.cps.CpsScript
+
+class JUnitPipelineStep implements Serializable {
+    CpsScript script
+
+    JUnitPipelineStep(CpsScript script) {
+        this.script = script
+    }
+
+    def call(Map args) {
+        String testResults = args.containsKey("testResults") ? args.testResults : ""
+        boolean keepLongStdio = args.containsKey("keepLongStdio") ? args.keepLongStdio : false
+        boolean allowEmptyResults = args.containsKey("allowEmptyResults") ? args.allowEmptyResults : false
+        Double healthScaleFactor = args.containsKey("healthScaleFactor") ? args.healthScaleFactor : 1.0
+
+        script.step($class: "JUnitResultArchiver",
```

Comment:
:-1:, this adds no value at all, other than creating a one-off DSL with no `Snippetizer` support that will be obsoleted by JENKINS-29922.